rnsdomains/rns-manager-react
# Failed Transaction Fallback and Estimation of Gas for Registration

# Exploring Gas Issues

Adds fallback during registration if the transaction fails at the miner level. Puts the user back to the previous step. Second, the estimation of gas cost for both registration transactions.

Rebased to Develop

Missing Translations

## Testing Instructions

These steps should be tested with both Nifty and Metamask on Chrome and Firefox. If testing on Local set the mining interval instead of automine.

1) request to register a domain
2) Change the gas limit to `26000`
3) **Expect** the transaction to fail and it shows a message, you are back at the beginning of step 1.
4) Don't change the gas limit and let it choose it for you
5) **Expect** transaction to succeed and move to step 2.
6) Step 2: Domain is ready to be registered, click on "register"
7) **Expect** the suggested gas amount to be much higher 
8) Again, set the gas limit low, to `26000`
9) **Expect** to see an error and back at step 2.
10) Leave the gas limit alone and proceed
11) **Expect** transaction to be successful
